### ISSUE TITLE
Closing the underlying core.async channel doesn't cause coerced source to become drained

### ISSUE THREAD

**Issue by ovan** [ztellman/manifold#53]
When a Manifold source is created by coercing a core.async channel with ->source it doesn't become drained when the channel is closed. This behavior differs from what happens with connect and therefore doesn't feel intuitive to me. Could this be a bug or is there some reason why this is actually the intended behavior?

```clojure
> (require `[manifold.source :as s])
nil
> (require '[clojure.core.async :as a])
nil

> (def ch (a/chan))
#'ch
> (def source (s/->source ch))
#'source
> (a/close! ch)
nil
> (s/drained? source)
false
> (s/take! source ::drained)
<< ::drained >>
```

```clojure
> (def ch (a/chan))
#'ch
> (def stream (s/stream))
#'stream
> (s/connect ch stream)
nil
> (a/close! ch)
nil
> (s/drained? stream)
true
> (s/take! stream ::drained)
<< ::drained >>
```

**Comment by ovan** [comment]
The behavior seems to be the same with both manifold 0.1.0 and 0.1.1-alpha4

**Comment by ztellman** [comment]
Since there's no way to know if a channel is drained until we fail to take from it, `drained?` will only return true after the last REPL entry in your example.  However, when trying this out I noticed that some code paths don't do this, so I pushed a change that fixes that.

Let me know if you have any questions.

**Comment by ovan** [comment]
That makes perfect sense. The example was artificial and only an attempt to reproduce minimally. I ran into this when a source that I expose from public API didn't hit the drained callback which is the important thing in my case. I worked around it by using connect in place of ->source. Great that this is fixed in 0.1.1. Thanks!
